src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -561,43 +563,6 @@ R"(#include <stdio.h>
         src = out;
     }
 
-    void visit_Cast(const ASR::Cast_t &x) {
-        self().visit_expr(*x.m_arg);
-        switch (x.m_kind) {
-            case (ASR::cast_kindType::IntegerToReal) : {
-                src = "(float)(" + src + ")";
-                break;
-            }
-            case (ASR::cast_kindType::RealToInteger) : {
-                src = "(int)(" + src + ")";
-                break;
-            }
-            case (ASR::cast_kindType::RealToReal) : {
-                // In C++, we do not need to cast float to float explicitly:
-                // src = src;
-                break;
-            }
-            case (ASR::cast_kindType::IntegerToInteger) : {
-                // In C++, we do not need to cast int <-> long long explicitly:
-                // src = src;
-                break;
-            }
-            case (ASR::cast_kindType::ComplexToComplex) : {
-                break;
-            }
-            case (ASR::cast_kindType::ComplexToReal) : {
-                src = "creal(" + src + ")";
-                break;
-            }
-            case (ASR::cast_kindType::LogicalToInteger) : {
-                src = "(int)(" + src + ")";
-                break;
-            }
-            default : throw CodeGenError("Cast kind " + std::to_string(x.m_kind) + " not implemented");
-        }
-        last_expr_precedence = 2;
-    }
-
```

Comment:
Why did you remove these? 

Is it needed for this PR? If not, let's do it separately. I am hoping to have as much in the common visitor as possible, but it looks like currently they are overridden in the subclasses (C and C++ visitors)?

Reply:
Okay, will change that in new PR.